### PR TITLE
ZipKin Tracing

### PR DESCRIPTION
**Pull request by PurpleBooth**
Motivation and Context
==

Currently there's no way to trace the requests through the application.

# What Changed

This will add the tracing headers to any request that comes through this
application onto another service, or preserve existing headers allowing the user
to follow requests through the application as a whole.

There are also new configuration settings:

```
| ZIPKIN_DISABLE                  | Totally disable Zipkin (including tracing headers)            | False
| ZIPKIN_DSN                      | Zipkin Sample API URL (e.g. http://zipkin:9411/api/v1/spans)  | None
| ZIPKIN_SAMPLE_RATE              | Percentage of requests to send to zipkin span API             | 0
```

By default this will add the tracing headers, but make no requests to Zipkin's neat little UI.

# How to test

No visible functional changes, check to see if downstream services are receiving
the headers: 

* 'X-B3-TraceId'
* 'X-B3-SpanId'
* 'X-B3-ParentSpanId'
* 'X-B3-Flags'
* 'X-B3-Sampled'

You can also set the environment variables

To see the spans go to zipkin:

* Set the environment variable `ZIPKIN_DSN` to a locally running zipkin server
* Set the environment variable `ZIPKIN_SAMPLE_RATE` to 100

Then run the acceptance tests.

# Links
* [Add Call time metrics into Splunk for Python Services (3d)][tkt]
* https://github.com/ONSdigital/ras-collection-instrument/pull/126
* https://github.com/ONSdigital/ras-party/pull/151
* https://github.com/ONSdigital/ras-secure-message/pull/227
* https://github.com/ONSdigital/response-operations-ui/pull/219
* https://github.com/ONSdigital/ras-frontstage/pull/383
* https://github.com/ONSdigital/rm-collection-exercise-service/pull/114
* https://github.com/ONSdigital/rm-case-service/pull/71
* https://github.com/ONSdigital/rm-actionexporter-service/pull/46
* https://github.com/ONSdigital/iac-service/pull/20
* https://github.com/ONSdigital/rm-sample-service/pull/74
* https://github.com/ONSdigital/rm-notify-gateway/pull/44
* https://github.com/ONSdigital/rm-sdx-gateway/pull/20
* https://github.com/ONSdigital/ras-party/pull/155
* https://github.com/ONSdigital/ras-rm-docker-dev/pull/80

[tkt]: https://trello.com/c/eY86bZG9/83-add-call-time-metrics-into-splunk-for-python-services-3d

### CODE REVIEW

**Comment by PurpleBooth** [comment]
[Add Call time metrics into Splunk for Python Services (3d)](https://trello.com/c/eY86bZG9/83-add-call-time-metrics-into-splunk-for-python-services-3d)

File: config.py [hunk 1]
```diff
@@ -21,7 +24,12 @@ class Config(object):
     GOOGLE_TAG_MANAGER = os.getenv('GOOGLE_TAG_MANAGER', None)
     NON_DEFAULT_VARIABLES = ['SECRET_KEY', 'SECURITY_USER_NAME', 'SECURITY_USER_PASSWORD', 'JWT_SECRET']
     AVAILABILITY_BANNER = os.getenv('AVAILABILITY_BANNER', False)
-    
+
+    # Zipkin
+    ZIPKIN_DISABLE = bool(strtobool(os.getenv("ZIPKIN_DISABLE", "False")))
```
Comment (tej99) [inline config.py]: Doesn't look like it's being used anywhere :/ unless there's some magic going on in which case...

We don't want it disabled by default? Does the docker-dev and ras-deloy configs need changing to disable this?

Comment (PurpleBooth) [reply config.py]: This is being used in the Zipkin library https://github.com/qiajigou/flask-zipkin/blob/master/flask_zipkin.py#L68

Comment (PurpleBooth) [reply config.py]: Disabling it will remove the zipkin headers from requests as well as sending messages to zipkin

**Pull request merged**